packages/core/utils/lib/convert-query-params.js

```diff
@@ -300,7 +336,7 @@ const convertAndSanitizeFilters = (filters, schema) => {
 
   // Here, `key` can either be an operator or an attribute name
   for (const [key, value] of Object.entries(filters)) {
-    const attribute = get(key, schema.attributes);
+    const attribute = key === 'id' ? { type: 'integer' } : get(key, schema.attributes);
```

Comment:
why is this change needed?

Reply:
If it is that way, my question would be (more for @alexandrebodin ?), why the `id` is not in the schema attributes.

Reply:
Yep, it's for the filter by ID functionality!
And the ID is not in the attributes because [the spec](https://jsonapi.org/format/) says it has to be a top-level field.

Reply:
oh! makes sense! Thank you @WalkingPizza  😃